### PR TITLE
Create DeleteWait and DeleteTimeout parameters for Destroy

### PR DESCRIPTION
**Pull request by virginiabrioso**
I noticed that when using wait: true the flag "--wait" was only being implemented in the sync/apply commands, this commit is to extend this to the destroy/delete commands

Usage:

`helmfile destroy` together with

```
helmDefaults:
  deleteWait: true
  deleteTimeout: 600
```

that will be equivalent to 

`helm uninstall --wait --timeout 600`

Related to https://github.com/helmfile/helmfile/issues/1057

[Create a new PR because the last one was a bit confusing]

### CODE REVIEW

File: pkg/state/state.go [hunk 1]
```diff
@@ -2066,7 +2090,18 @@ func (st *HelmState) DeleteReleases(affectedReleases *AffectedReleases, helm hel
...
+			timeout := st.HelmDefaults.DeleteTimeout
+			if release.DeleteTimeout != nil {
+				timeout = *release.DeleteTimeout
+			}
+			if timeout != 0 {
+				duration := strconv.Itoa(timeout)
+				duration += "s"
+				flags = append(flags, "--timeout", duration)
+			}
+		}
```
Comment (yxxhero) [inline pkg/state/state.go:2104]: same as above. please DRY.

File: pkg/state/state.go [hunk 2]
```diff
@@ -797,6 +806,18 @@ func (st *HelmState) DeleteReleasesForSync(affectedReleases *AffectedReleases, h
...
+						timeout := st.HelmDefaults.DeleteTimeout
+						if release.DeleteTimeout != nil {
+							timeout = *release.DeleteTimeout
+						}
+						if timeout != 0 {
+							duration := strconv.Itoa(timeout)
+							duration += "s"
+							args = append(args, "--timeout", duration)
+						}
+					}
```
[the remark this answers is not in the document]
Comment (virginiabrioso) [reply pkg/state/state.go]: Done

File: pkg/state/state.go (hunk 1 above)
Comment (virginiabrioso) [reply pkg/state/state.go]: Done

File: docs/index.md [hunk 3]
```diff
@@ -221,6 +221,10 @@ helmDefaults:
   cascade: "background"
   # insecureSkipTLSVerify is true if the TLS verification should be skipped when fetching remote chart
   insecureSkipTLSVerify: false
+  # --wait flag for destroy/delete, if set to true, will wait until all resources are deleted before mark delete command as successful
+  deleteWait: false
+  # Timeout is the time in seconds to wait for helmfile destroy/delete (default 300)
+  deleteWait: 300
```
Comment (yxxhero) [inline docs/index.md:227]: double deleteWait?

File: cmd/destroy.go [hunk 4]
```diff
@@ -35,6 +35,7 @@ func NewDestroyCmd(globalCfg *config.GlobalImpl) *cobra.Command {
 	f.StringVar(&destroyOptions.Cascade, "cascade", "", "pass cascade to helm exec, default: background")
 	f.IntVar(&destroyOptions.Concurrency, "concurrency", 0, "maximum number of concurrent helm processes to run, 0 is unlimited")
 	f.BoolVar(&destroyOptions.SkipCharts, "skip-charts", false, "don't prepare charts when destroying releases")
+	f.BoolVar(&destroyOptions.DeleteWait, "deleteWait", false, `override helmDefaults.wait setting "helm uninstall --wait"`)
```
Comment (yxxhero) [inline cmd/destroy.go:38]: where si the DeleteTimeout?

File: pkg/state/state.go [hunk 5]
```diff
@@ -800,6 +825,7 @@ func (st *HelmState) DeleteReleasesForSync(affectedReleases *AffectedReleases, h
 					if release.Namespace != "" {
 						args = append(args, "--namespace", release.Namespace)
 					}
+					args = st.DeleteWait(args, release)
```
Comment (yxxhero) [inline pkg/state/state.go:828]: appendDeleteWaitFlags

File: pkg/state/state.go [hunk 6]
```diff
@@ -765,6 +774,22 @@ func ReleaseToID(r *ReleaseSpec) string {
 	return id
 }
 
+func (st *HelmState) DeleteWait(args []string, release *ReleaseSpec) []string {
```
Comment (yxxhero) [inline pkg/state/state.go:777]: could you add some unittests for this func?

File: docs/index.md (hunk 3 above)
Comment (virginiabrioso) [reply docs/index.md:227]: Done

File: pkg/state/state.go (hunk 6 above)
Comment (virginiabrioso) [reply pkg/state/state.go:777]: I added to cases into pkg/state/state_test.go

**Comment by felipecrs** [comment]
I have been testing this PR over the weekend and it worked fine btw. :)

**Comment by yxxhero** [comment]
@virginiabrioso LGTM

**Review by yxxhero — APPROVED** [review]

**Pull request merged**